### PR TITLE
Integrate replica sync task and `PreferredSequencer`

### PR DESCRIPTION
**Pull request by bkolad**
# Description

This PR includes the following updates:
1. Renamed replica-related messages from `DoXX` to `ReplicaXX`.
2. Completed the implementation of `SequencerStateUpdator::on_db_event`. Replica-specific state update methods are now fully integrated.
3. Added handling for the `PreferredSeqOperation::Unreachable` case, which is now applicable to replicas. 

- [x] ~I have updated `CHANGELOG.md` with a new entry if my PR makes any breaking changes or fixes a bug. If my PR removes a feature or changes its behavior, I provide help for users on how to migrate to the new behavior.~
- [x] I have carefully reviewed all my `Cargo.toml` changes before opening the PRs. (Are all new dependencies necessary? Is any module dependency leaked into the full-node (hint: it shouldn't)?)

### CODE REVIEW

File: crates/full-node/sov-sequencer/src/preferred/inner.rs [hunk 1]
```diff
@@ -1716,7 +1738,15 @@ where
             condition_are_there_batches_to_replay,
             condition_node_is_unsynced_and_doesnt_know_it,
         ) {
-            (true, _, _, true, _) => PreferredSeqOperation::Unreachable,
+            (true, _, _, true, _) => {
```
Comment (bkolad) [inline crates/full-node/sov-sequencer/src/preferred/inner.rs:1741]: We’ll also need to handle other edge cases — for example, clarifying what it means for a replica to recover. These will be addressed in upcoming PRs.

File: examples/demo-rollup/tests/replica/mod.rs [hunk 2]
```diff
@@ -154,21 +154,9 @@ async fn test_replica_receives_txs_from_postgres() {
...
-        tokio::time::sleep(std::time::Duration::from_millis(
-            sov_test_utils::TEST_DEFAULT_MOCK_DA_BLOCK_TIME_MS,
-        ))
-        .await;
-    }
 
-    test_rollup
-        .wait_for_height(height_before_tx.get() + gap)
-        .await;
+    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```
Comment (bkolad) [inline examples/demo-rollup/tests/replica/mod.rs:159]: In this test, block production is done manually, and we’ve replaced block creation with a timeout. As a result, the replica can only observe updates through PostgreSQL. This test verifies that it behaves as intended.

File: crates/full-node/sov-sequencer/src/preferred/inner.rs [hunk 3]
```diff
@@ -1989,7 +2006,17 @@ where
...
 
+        let recovery_executor = inner.new_executor_with_empty_uncommitted_changes(&info);
+
+        inner
+            .force_overwrite_state(info.clone(), recovery_executor)
+            .await;
+
         inner.update_api_ledger(&info).await;
+
+        if inner.is_replica() && info.sync_status.distance() <= distance {
```
Comment (theodorebugnet) [inline crates/full-node/sov-sequencer/src/preferred/inner.rs:2017]: What's the reason for special-casing this here? IIRC, the master sequencer will wait for an extra state update before setting ready = Ok(). Are there any issues with having replicas use the same codepath?

**Review by theodorebugnet — APPROVED** [review]

File: crates/full-node/sov-sequencer/src/preferred/inner.rs (hunk 3 above)
Comment (bkolad) [reply crates/full-node/sov-sequencer/src/preferred/inner.rs:2017]: Yes, without this, the replica was entering an infinite resync loop. The issue is that after the extra state update, the inner executor still lags behind the node executor (so replica continues resyncing). If we mark it as OK here, we can start applying the PG transactions on the inner executor, which then races ahead of the node. I’ll look into making this cleaner.
